packages/trace-viewer/src/ui/modelUtil.ts

```diff
@@ -160,6 +160,56 @@ function indexModel(context: ContextEntry) {
 }
 
 function mergeActionsAndUpdateTiming(contexts: ContextEntry[]) {
+  const traceFileToContexts = new Map<string, ContextEntry[]>();
+  for (const context of contexts) {
+    const traceFile = context.traceUrl;
+    let list = traceFileToContexts.get(traceFile);
+    if (!list) {
+      list = [];
+      traceFileToContexts.set(traceFile, list);
+    }
+    list.push(context);
+  }
+
+  const result: ActionTraceEventInContext[] = [];
+  let traceFileId = 0;
+  for (const [, contexts] of traceFileToContexts) {
+    // Action ids are unique only within a trace file. If there are
+    // traces from more than one file we make the ids unique across the
+    // files. The code does not update snapshot ids as they are always
+    // retrieved from a particular trace file.
+    if (traceFileToContexts.size)
+      makeCallIdsUniqueAcrossTraceFiles(contexts, ++traceFileId);
+    // Align action times across runner and library contexts within each trace file.
+    const map = mergeActionsAndUpdateTimingSameTrace(contexts);
+    result.push(...map.values());
+  }
+  result.sort((a1, a2) => {
+    if (a2.parentId === a1.callId)
+      return -1;
+    if (a1.parentId === a2.callId)
+      return 1;
+    return a1.wallTime - a2.wallTime || a1.startTime - a2.startTime;
```

Comment:
Should we sort by `startTime` only? Maybe in the next patch?

Reply:
Let me tackle it in one of the next patches where I change how the times are aligned.